public/app/plugins/datasource/tempo/streaming.ts

```diff
@@ -0,0 +1,77 @@
+import { map, Observable, defer, mergeMap } from 'rxjs';
+import { v4 as uuidv4 } from 'uuid';
+
+import {
+  DataFrame,
+  DataQueryRequest,
+  DataQueryResponse,
+  DataSourceInstanceSettings,
+  LiveChannelScope,
+  LoadingState,
+  MutableDataFrame,
+} from '@grafana/data';
+import { getGrafanaLiveSrv } from '@grafana/runtime';
+
+import { SearchStreamingState } from './dataquery.gen';
+import { TempoDatasource } from './datasource';
+import { createTableFrameFromTraceQlQuery } from './resultTransformer';
+import { TempoJsonData, TempoQuery } from './types';
+export async function getLiveStreamKey(query: TempoQuery): Promise<string> {
+  return uuidv4();
+}
+
+export function doTempoChannelStream(
+  query: TempoQuery,
+  ds: TempoDatasource,
+  options: DataQueryRequest<TempoQuery>,
+  instanceSettings: DataSourceInstanceSettings<TempoJsonData>
+): Observable<DataQueryResponse> {
+  const range = options.range;
+
+  let frames: DataFrame[] | undefined = undefined;
+  let state: LoadingState = LoadingState.NotStarted;
+
+  return defer(() => getLiveStreamKey(query)).pipe(
+    mergeMap((key) => {
+      return getGrafanaLiveSrv()
+        .getStream<MutableDataFrame>({
+          scope: LiveChannelScope.DataSource,
+          namespace: ds.uid,
+          path: `search/${key}`,
```

Comment:
Btw this does not create a separate websocket connection right? It just subscribes to a channel over single websocket connection?